### PR TITLE
Lookup Table and Template Builder base documentation

### PR DESCRIPTION
**Pull request by nmadruga**
Add new basic documentation for Lookup tables and images for Template builder.

- Lookup tables was based on existing documentation 
  - New page created and linked in Home
  - ~~just need to add the images (updated)~~ DONE
- Template builder was missing this overview
  - I got a little carried away and explained a bit more of fields edition... It would be helpful to have this included though!
  - TODO: There is one example missing images too from another existing documentation - I'll leave this for the future.

### CODE REVIEW

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:88]: Also -- this hasn't been auto-formatted either -- it should have converted the "*" into "-"

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:92]: Missing paragraph break

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:94]: Link to core actions: https://github.com/openmsupply/conforma-server/wiki/List-of-Action-plugins#core-actions

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:96]:
```suggestion
The Actions tab displays (and defines) any action behaviour that is specific to this application type that will run on the specified Trigger events.
```

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:97]: Case: `sendNotification`

Also, there's an extra ")" in that link that makes it not render correctly.

Comment (CarlosNZ) [reply docs/internal/config/Template-Builder.md:97]: Also missing paragraph break

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:101]: Should be level 2 "##" heading

**Review by CarlosNZ — APPROVED** [review]
Cool, thanks for doing this. I'll approve now so you can merge when you're ready, but there's just a bunch of small typos and suggestions to have a look at.

Main thing is I don't think your Markdown extension is auto-formatting correctly. You might have it disabled? When I resave these files in my system, they get auto-formatted correctly (as per the rest of the documentation)

Also -- maybe the Template Builder page could do with a Table of contents at the top (see other files for how to auto-generate)

File: docs/internal/config/Lookup-Tables.md
Comment (nmadruga) [reply docs/internal/config/Lookup-Tables.md:1]: Cool - would you like me to go ahead and remove from the other existing pages that contains a Heading like that one?

**Comment by nmadruga** [comment]
> but there's just a bunch of small typos and suggestions to have a look at.

Thanks for looking into this Carl. I just wrote and didn't have a proper review. So will look now at your comments and fix the suggestions, then add the missing images before I merge this in and update the wiki!

**Comment by nmadruga** [comment]
> Main thing is I don't think your Markdown extension is auto-formatting correctly. You might have it disabled? When I resave these files in my system, they get auto-formatted correctly (as per the rest of the documentation)

I noticed it wasn't working when I created a table! I'll also check this before committing latest changes

File: docs/internal/config/Template-Builder.md
Comment (nmadruga) [reply docs/internal/config/Template-Builder.md:81]: Oh! That's right. I was assuming something, but will remove now. I see it's not there. Maybe I'll just mention that there are compulsory and optional parameters for each Plugin type.

**Comment by nmadruga** [comment]
All done! And already updated to the Wiki

Anyone would you like to have a final look @CarlosNZ, @jagoje and @fergie-nz ?

File: docs/internal/Filters-for-Applications.md
Comment (CarlosNZ) [inline docs/internal/Filters-for-Applications.md:7]: Don't change it here, but in the place where its used -- should be capitalised "Applicant action" for consistency with the others.

File: docs/internal/config/List-Core-Actions.md
Comment (CarlosNZ) [inline docs/internal/config/List-Core-Actions.md:13]: There's duplication here of the "core action" list (and it's out of date). Can you just replace that section with a link to https://github.com/openmsupply/conforma-server/wiki/List-of-Action-plugins#core-actions ?

File: docs/internal/config/Lookup-Tables.md
Comment (CarlosNZ) [inline docs/internal/config/Lookup-Tables.md]: Typo "Loopup" ;)

Comment (CarlosNZ) [reply docs/internal/config/Lookup-Tables.md]: And below too

File: docs/internal/config/Lookup-Tables.md
Comment (CarlosNZ) [inline docs/internal/config/Lookup-Tables.md]: None of the images aren't showing up correctly. This is not the correct image formatting I don't think. (Check another one where it works).

File: docs/internal/config/Lookup-Tables.md
Comment (CarlosNZ) [inline docs/internal/config/Lookup-Tables.md]: If you want these to go within the list, you need to use proper Markdown line breaks (double space at end of the line)

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:51]: Typo: serial_patern

Also, I'm sure I mentioned this in the last review, but seems to be gone now...

Instead of "- used regular expression" it should say something like:
```
uses "Generate Text String" action (with link to https://github.com/openmsupply/conforma-server/wiki/List-of-Action-plugins#generate-text-string)

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:57]: These two should mention that value can be an evaluator expression

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:61]: "applicant's" should be plural "applicants", not possessive  :)

**Review by CarlosNZ — CHANGES_REQUESTED** [review]

File: docs/internal/config/Template-Builder.md
Comment (nmadruga) [reply docs/internal/config/Template-Builder.md:61]: hahah that's right! English is though

Comment (CarlosNZ) [reply docs/internal/config/Template-Builder.md:61]: > English is ~~though~~ **tough**

😆

Yes, indeed!

File: docs/internal/config/Template-Builder.md
Comment (nmadruga) [reply docs/internal/config/Template-Builder.md:57]: Can it? I though it was just Markdown

File: docs/internal/config/Template-Builder.md
Comment (nmadruga) [reply docs/internal/config/Template-Builder.md:51]: Thanks! I'll add this link

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [reply docs/internal/config/Template-Builder.md:57]: > Can it? I though it was just Markdown

Yes, definitely.

Screenshot 2022-12-15 at 4 52 46 PM

File: docs/internal/config/List-Core-Actions.md
Comment (nmadruga) [reply docs/internal/config/List-Core-Actions.md:13]: Yes - for sure! Thanks for suggesting

File: docs/internal/config/Template-Builder.md
Comment (nmadruga) [reply docs/internal/config/Template-Builder.md]: Thanks! Added. I think all has been addressed now (the Documentation in Wiki is also updated!)

Should we merged this one now @CarlosNZ ?

File: docs/internal/config/Template-Builder.md
Comment (CarlosNZ) [inline docs/internal/config/Template-Builder.md:56]: "A Evaluator Expression" => "An Evaluator expression" X 2 ;)

**Review by CarlosNZ — APPROVED** [review]
👍

**Pull request merged**
